app/controllers/planning_applications/review/recommendations_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module PlanningApplications
+  module Review
+    class RecommendationsController < AuthenticationController
+      include CommitMatchable
+
+      before_action :set_planning_application
+      before_action :ensure_user_is_reviewer_checking_assessment, only: %i[update edit]
+      before_action :set_recommendations, only: %i[update edit]
+      before_action :set_committee_decision
+      before_action :set_recommendation, only: :update
+
+      def new
+        @recommendation = Recommendation.new
+      end
+
+      def edit
+        @recommendation = @recommendations.last
+
+        respond_to do |format|
+          format.html
+        end
+      end
+
+      def create
+        @recommendation = Recommendation.new(
+          planning_application: @planning_application.planning_application,
+          status: "review_complete",
+          reviewer: Current.user
+        )
+
+        if @recommendation.save
+          @planning_application.update!(decision: recommendation_form_params[:decision], public_comment: recommendation_form_params[:public_comment])
+          @planning_application.submit!
+          redirect_to planning_application_review_tasks_path(@planning_application), notice: t(".success")
+        else
+          render :new
+        end
+      end
+
+      def update
+        respond_to do |format|
+          format.html do
+            @recommendation.assign_attributes(recommendation_params)
+            render :edit and return unless @recommendation.valid?
+
+            if @recommendation.review_complete?
+              @recommendation.review!
+            else
+              @recommendation.save!
+            end
+
+            if @recommendation.challenged?
+              if @recommendation.committee_overturned?
+                redirect_to new_planning_application_review_recommendation_path(@planning_application)
+              else
+                redirect_to planning_application_review_tasks_path(@planning_application),
+                  notice: t(".success")
+              end
+            else
+              redirect_to planning_application_review_tasks_path(@planning_application),
+                notice: t(".success")
+            end
+          end
```

Comment:
Again move all the model logic to the model class and then extract the redirect path logic to a private method so you can do something like:

``` ruby
if @recommendation.save_and_review(recommendation_params)
  redirect_to after_save_and_review_location_url, notice: t(".success")
else
  render :edit
end
```